core/types/transaction.go

```diff
@@ -578,6 +578,78 @@ func (tx *Transaction) WithSignature(signer Signer, sig []byte) (*Transaction, e
 	return &Transaction{inner: cpy, time: tx.time}, nil
 }
 
+// IntrinsicGas returns the calculated intrinsic gas for the given transaction.
+func (tx *Transaction) IntrinsicGas(rules *params.Rules) (uint64, error) {
+	return calcIntrinsicGas(tx.inner, rules)
+}
+
+// IntrinsicGas returns the calculated intrinsic gas for the given transaction.
+func IntrinsicGas(txdata TxData, rules *params.Rules) uint64 {
+	gas, err := calcIntrinsicGas(txdata, rules)
+	if err != nil {
+		panic(err)
```

Comment:
Isn't panicing a bit harsh here? I know its only used in tests and the rpc, maybe we could just return maxUint64 or an error